File: CORREDOR/4 Lenguajes/JS/TiposDatos.js
class TiposDatos {
    static main() {
        // 🔢 1. Tipos numéricos enteros
        let edad = 25;
        let año = 2025;
        let poblacion = 500000;
        let distancia = 9876543210;
        let letra = 'A';

        console.log("Tipos Enteros");
        console.log("Edad: " + edad);
        console.log("Año: " + año);
        console.log("Población: " + poblacion);
        console.log("Distancia: " + distancia);
        console.log("Letra: " + letra);

        let precio = 19.99;
        let peso = 70.45;
        console.log("Tipos Decimales");
        console.log("Precio: " + precio);
        console.log("Peso: " + peso);

        let nombre = "Johan";
        let saludo = "Hola, ¿cómo estás?";
        let curso = "Programación en JavaScript";

        console.log("Tipo Texto");
        console.log(nombre);
        console.log(saludo);
        console.log(curso);

        let esMayorDeEdad = true;
        let tieneLicencia = false;
        let aprobado = true;
        let encendido = false;
        let accesoPermitido = true;

        console.log("Tipo Booleano");
        console.log("Mayor de edad: " + esMayorDeEdad);
        console.log("Tiene licencia: " + tieneLicencia);
        console.log("Aprobado: " + aprobado);
        console.log("Encendido: " + encendido);
        console.log("Acceso permitido: " + accesoPermitido);

        // 💾 5. Tipo binario (representado con números o texto)
        let binario1 = 0b1010;
        let binario2 = 0b1100;
        let binario3 = 0b1111;
        let binario4 = 0b100000;
        let binario5 = "101010";

        console.log("Tipo Binario");
        console.log("Binario 1: " + binario1);
        console.log("Binario 2: " + binario2);
        console.log("Binario 3: " + binario3);
        console.log("Binario 4: " + binario4);
        console.log("Binario 5 (texto): " + binario5);
    }
}

TiposDatos.main();

File: CORREDOR/4 Lenguajes/Java/TiposDatos.java
package Java;

public class TiposDatos {
    public static void main(String[] args) {

        // 🔢 1. Tipos numéricos enteros
        byte edad = 25;                
        short año = 2025;               
        int poblacion = 500000;         
        long distancia = 9876543210L;   
        char letra = 'A';               

        System.out.println("Tipos Enteros");
        System.out.println("Edad: " + edad);
        System.out.println("Año: " + año);
        System.out.println("Población: " + poblacion);
        System.out.println("Distancia: " + distancia);
        System.out.println("Letra: " + letra);

        // 🔢 2. Tipos numéricos decimales
        float precio = 19.99f;   
        double peso = 70.45;     
        System.out.println("Tipos Decimales");
        System.out.println("Precio: " + precio);
        System.out.println("Peso: " + peso);

        // 💬 3. Tipo texto (String)
        String nombre = "Johan";
        String saludo = "Hola, ¿cómo estás?";
        String curso = "Programación en Java";
        String ciudad = "Bogotá";
        String frase = "El conocimiento es poder";

        System.out.println("Tipo Texto");
        System.out.println(nombre);
        System.out.println(saludo);
        System.out.println(curso);
        System.out.println(ciudad);
        System.out.println(frase);

        // ✅ 4. Tipo booleano (true / false)
        boolean esMayorDeEdad = true;
        boolean tieneLicencia = false;
        boolean aprobado = true;
        boolean encendido = false;
        boolean accesoPermitido = true;

        System.out.println("Tipo Booleano");
        System.out.println("Mayor de edad: " + esMayorDeEdad);
        System.out.println("Tiene licencia: " + tieneLicencia);
        System.out.println("Aprobado: " + aprobado);
        System.out.println("Encendido: " + encendido);
        System.out.println("Acceso permitido: " + accesoPermitido);

        // 💾 5. Tipo binario (representado con int o byte)
        byte binario1 = 0b1010;     
        byte binario2 = 0b1100;     
        int binario3 = 0b1111;       
        int binario4 = 0b100000;     
        String binario5 = "101010";  

        System.out.println("Tipo Binario");
        System.out.println("Binario 1: " + binario1);
        System.out.println("Binario 2: " + binario2);
        System.out.println("Binario 3: " + binario3);
        System.out.println("Binario 4: " + binario4);
        System.out.println("Binario 5 (texto): " + binario5);
    }
}